js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import GridiconPlusSmall from 'gridicons/dist/plus-small';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Section from '.~/wcdl/section';
+import AppButton from '.~/components/app-button';
+import AppButtonModalTrigger from '.~/components/app-button-modal-trigger';
+import VerticalGapLayout from '.~/components/vertical-gap-layout';
+import { SHIPPING_RATE_METHOD } from '.~/constants';
+import MinimumOrderInputControl from './minimum-order-input-control';
+import AddMinimumOrderModal from './add-minimum-order-modal';
+import groupShippingRatesByFreeShippingThreshold from './groupShippingRatesByFreeShippingThreshold';
+import './minimum-order-card.scss';
+
+const MinimumOrderCard = ( props ) => {
+	const { value = [], onChange = noop } = props;
+	const nonZeroShippingRates = value.filter(
+		( shippingRate ) =>
+			shippingRate.rate > 0 &&
+			shippingRate.method === SHIPPING_RATE_METHOD.FLAT_RATE
+	);
```

Comment:
Made a code refactor in [`b20ede6` (#1266)](https://github.com/woocommerce/google-listings-and-ads/pull/1266/commits/b20ede66be05419a5b24e815a540a77e73f7ed7a), with a new function called `isNonZeroFlatShippingRate` (not sure if you have a better naming? I don't. 😂 )

Reply:
> new function called `isNonZeroFlatShippingRate` (not sure if you have a better naming? I don't. 😂 )

No idea either. 😂 

And I'm slightly more leaning towards the below approach:

```js
// util
function getChargedShippingRates( rates, method = SHIPPING_RATE_METHOD.FLAT_RATE ) {
    return rates.filter(
        ( shippingRate ) =>
            shippingRate.rate > 0 && shippingRate.method === method
        );
}

// <MinimumOrderCard>
const nonZeroShippingRates = getChargedShippingRates( value );

// <FlatShippingRatesInputCards>
const displayFreeShippingCards = getChargedShippingRates( value ).length > 0;
```
Even though all countries were selected, there should be no significant difference in the calculation performance here.

No matter which way, it's already better. Thanks for the adjustment!

Reply:
> ```js
> getChargedShippingRates
> ```

You gave me a better naming idea, I have changed the function name from `isNonZeroFlatShippingRate` to `isNonFreeFlatShippingRate` (also based on wiktionary page on "[nonfree](https://en.wiktionary.org/wiki/nonfree)"), in [`85e0033` (#1266)](https://github.com/woocommerce/google-listings-and-ads/pull/1266/commits/85e00336bcba72cc8b3e7c5a51fada4e030c00b5). 😄 

> And I'm slightly more leaning towards the below approach

I still prefer the current `isNonFreeFlatShippingRate` way, because the function is smaller, more reusable and more readable, in my opinion. 💅 😄 🙏 

```js
// <MinimumOrderCard>
const nonZeroShippingRates = value.filter( isNonFreeFlatShippingRate );

// <FlatShippingRatesInputCards>:
const displayFreeShippingCards = value.some( isNonFreeFlatShippingRate );

// And we can use it if we are dealing with one shipping rate.
const isNonFreeFlat = isNonFreeFlatShippingRate( shippingRate )
```